controllers/extension-certificate-service/pkg/controller/certservice/actuator.go

```diff
@@ -0,0 +1,335 @@
+	if err != nil {
+		return err
+	}
+	configValues := make(map[string]interface{})
+	release, err := chartRenderer.Render(
+		certBrokerChartPath,
+		certBrokerName,
+		namespace,
+		configValues)
+	if err != nil {
+		return fmt.Errorf("failed to render chart: %v", err)
+	}
+
+	var (
+		manifest   = release.Manifest()
+		decoder    = yaml.NewYAMLOrJSONDecoder(bytes.NewReader(manifest), 1024)
+		decodedObj map[string]interface{}
+	)
+
+	a.logger.Info("Component is being deleted", "component", "cert-broker", "namespace", namespace)
+	for err = decoder.Decode(&decodedObj); err == nil; err = decoder.Decode(&decodedObj) {
+		if decodedObj == nil {
+			continue
+		}
+
+		obj := unstructured.Unstructured{Object: decodedObj}
+		decodedObj = nil
+
+		// only delete namepaced resources
+		if len(obj.GetNamespace()) > 0 {
+			if err := a.client.Delete(ctx, &obj); err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+		}
+
+	}
+
+	secret, err := a.createKubeconfigForCertManager(ctx, namespace)
+	if err != nil {
+		return err
+	}
+
+	if err := a.client.Delete(ctx, secret); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("error cleaning up secret for rbac manager: %v", err)
+	}
+
+	return nil
+}
+
+var rbacChartPath = filepath.Join(utils.ChartsPath, "cert-broker-rbac")
+
+func (a *actuator) createRBAC(ctx context.Context, config *rest.Config) error {
+	applier, err := kubernetes.NewChartApplierForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create chart applier: %v", err)
+	}
+
+	a.logger.Info("Manifests are being applied")
+	if err := applier.ApplyChart(ctx, rbacChartPath, "", "", nil, nil); err != nil {
+		return fmt.Errorf("error applying manifests: %v", err)
+	}
+
+	return nil
+}
+
+func (a *actuator) deleteRBAC(ctx context.Context, config *rest.Config) error {
+	renderer, err := chartrenderer.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create chart renderer for rbac removal: %v", err)
+	}
+
+	release, err := renderer.Render(rbacChartPath, "", "", nil)
+	if err != nil {
+		return fmt.Errorf("failed to render charts for rbac removal: %v", err)
+	}
+
+	var (
+		manifest   = release.Manifest()
+		decoder    = yaml.NewYAMLOrJSONDecoder(bytes.NewReader(manifest), 1024)
+		decodedObj map[string]interface{}
+	)
+
+	a.logger.Info("Manifests are being deleted")
+	for err = decoder.Decode(&decodedObj); err == nil; err = decoder.Decode(&decodedObj) {
+		if decodedObj == nil {
+			continue
+		}
+
+		obj := unstructured.Unstructured{Object: decodedObj}
+		decodedObj = nil
+
+		targetClient, err := client.New(config, client.Options{})
+		if err != nil {
+			return fmt.Errorf("Cannot create client for target cluster to delete RBAC manifests: %v", err)
+		}
+
+		if err := targetClient.Delete(ctx, &obj); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (a *actuator) createKubeconfigForRBAC(ctx context.Context, namespace string) (*corev1.Secret, error) {
```

Comment:
As discussed: You can use the `gardener` secret/kubeconfig to do these things (https://github.com/gardener/gardener/blob/master/pkg/apis/core/v1alpha1/types_constants.go#L38)